script.js

```diff
@@ -1,12 +1,140 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+/* TODO LIST
+1 select all episoded
+2 if select and searche, should 
+3 counter
+4 rights@ for ....
+
+// let myData;
+
+// 9 scrip
+
+// window.onload = async function myDataEpisodes() {
+//   const response = await fetch(url);
+//   const data = await response.json();
+//   testing(data);
+// }
+*/
+
+//const url = "https://api.tvmaze.com/shows/82/episodes";
+
+window.onload = () => {
+  const allShows = getAllShows();
+  populateShowList(allShows);
+};
+
+function getEpisodes(show_id) {
+  fetch(`https://api.tvmaze.com/shows/${show_id}/episodes`)
+    .then(function (response) {
+      return response.json();
+    })
+    .then(function (data) {
+      testing(data);
+    });
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+// Global variables
+const container = document.querySelector("#root");
+const selectShow = document.querySelector("#shows_list_id");
+const select = document.querySelector("#select_id");
+const selectAllEpisodesOption = document.querySelector("#select_all");
+const countEpisodes = document.querySelector("#count_Episodes");
+const totalEpisodes = document.querySelector("#total_Episodes");
+const searchInput = document.querySelector("#input_search");
+
+function populateShowList(allShows) {
+  for (let show of allShows) {
+    const showsOption = document.createElement("option");
+    showsOption.setAttribute("value", show.id);
+    showsOption.innerText = show.name;
+    selectShow.appendChild(showsOption);
+  }
 }
 
-window.onload = setup;
+selectShow.addEventListener("change", (e) => {
+  value = e.target.value;
+  getEpisodes(value);
+});
+
+function testing(getData) {
+  const allEpisodes = getData;
+  totalEpisodes.innerText = ` - out of - ${allEpisodes.length} Episodes `;
+
+  searchInput.addEventListener("input", (e) => {
```

Comment:
Nice mets the requirements and all. Might clear the field when you select a diffrent show or a specfic episode :)